staging/src/k8s.io/apiserver/pkg/server/genericapiserver_test.go

```diff
@@ -682,3 +683,88 @@ func TestGracefulShutdown(t *testing.T) {
 		t.Errorf("Timed out waiting for response.")
 	}
 }
+
+func TestWarningWithRequestTimeout(t *testing.T) {
+	type result struct {
+		err        interface{}
+		stackTrace string
+	}
+	clientDoneCh, resultCh := make(chan struct{}), make(chan result, 1)
+	testHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// add warnings while we're waiting for the request to timeout to catch read/write races
+	loop:
+		for {
+			select {
+			case <-r.Context().Done():
+				break loop
+			default:
+				warning.AddWarning(r.Context(), "a", "1")
```

Comment:
this bit can trigger the panic, so move the defer that catches the panic for the test to the top of the function

Reply:
yes, thanks, i moved it hastily.